titan-compiler/checker.lua

```diff
@@ -2,32 +2,35 @@ local checker = {}
 
 local symtab = require 'titan-compiler.symtab'
 
+local function visit_children(f, node, ...)
+    local tag = node._tag
+    if not tag then -- the node is an array
+        for _, child in ipairs(node) do
+            f(child, ...)
+        end
+    else
+        for _, child in node:children() do
+            f(child, ...)
+        end
+    end
+end
+
```

Comment:
I think this smells funny. I created a [new issue](https://github.com/titan-lang/titan-v0/issues/17) to discuss if we should change this in a future commit.